eva-server/src/main/java/uk/ac/ebi/variation/eva/server/ws/ga4gh/GA4GHBeaconWSServer.java

```diff
@@ -19,51 +19,50 @@
 
 package uk.ac.ebi.variation.eva.server.ws.ga4gh;
 
-import io.swagger.annotations.Api;
 import java.io.IOException;
 import java.net.UnknownHostException;
 import java.util.Arrays;
-import javax.servlet.http.HttpServletRequest;
-import javax.ws.rs.GET;
-import javax.ws.rs.Path;
-import javax.ws.rs.Produces;
-import javax.ws.rs.QueryParam;
-import javax.ws.rs.core.Context;
-import javax.ws.rs.core.MediaType;
-import javax.ws.rs.core.Response;
-import javax.ws.rs.core.UriInfo;
+
+import javax.servlet.http.HttpServletResponse;
+
 import org.opencb.biodata.models.feature.Region;
+import org.opencb.datastore.core.QueryOptions;
 import org.opencb.datastore.core.QueryResult;
 import org.opencb.opencga.lib.auth.IllegalOpenCGACredentialsException;
 import org.opencb.opencga.storage.core.variant.adaptors.VariantDBAdaptor;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+import io.swagger.annotations.Api;
 import uk.ac.ebi.variation.eva.lib.datastore.DBAdaptorConnector;
 import uk.ac.ebi.variation.eva.server.ws.EvaWSServer;
 
 /**
  *
  * @author Cristina Yenyxe Gonzalez Garcia <cyenyxe@ebi.ac.uk>
  */
-@Path("/v1/ga4gh")
-@Produces(MediaType.APPLICATION_JSON)
+@RestController
+@RequestMapping(value = "/v1/ga4gh", produces = "application/json")
 @Api(tags = { "ga4gh" })
 public class GA4GHBeaconWSServer extends EvaWSServer {
 
-    public GA4GHBeaconWSServer(@Context UriInfo uriInfo, @Context HttpServletRequest hsr) {
-        super(uriInfo, hsr);
-    }
-
-    @GET
-    @Path("/beacon")
-    public Response beacon(@QueryParam("referenceName") String chromosome,
-                           @QueryParam("start") Integer start,
-                           @QueryParam("allele") String allele,
-                           @QueryParam("datasetIds") String studies) 
+    public GA4GHBeaconWSServer() { }
+    
+    @RequestMapping(value = "/beacon", method = RequestMethod.GET)
+    public GA4GHBeaconResponse beacon(@RequestParam("referenceName") String chromosome,
```

Comment:
By default, a 400 error and a message saying `required parameter X is missing` are automatically generated, unless said parameter is annotated as `required=false`.